usr/io.c

```diff
@@ -376,6 +376,27 @@ iscsi_io_tcp_connect(iscsi_conn_t *conn, int non_blocking)
 		}
 	}
 
+	/* optionally set the congestion control algo */
+	if (*conn->tcp_congestion) {
+		char congestion[sizeof(conn->tcp_congestion)];
+		socklen_t arglen = strlen(conn->tcp_congestion);
+
+		if (setsockopt(conn->socket_fd, IPPROTO_TCP, TCP_CONGESTION,
+		               conn->tcp_congestion, arglen) < 0) {
+			log_warning("failed to set TCP congestion control algo "
+				    "to %s", conn->tcp_congestion);
+		} else {
+			arglen = sizeof(congestion) - 1;
```

Comment:
Not exactly sure why you get the congestion setting after just successfully setting it. It looks like the only reason is for a debug log message, but that message only shows up if the *debug* log level is 4 or higher. It seems like a debug log message saying "setting congestion to BLAH worked", without having to actually get it again?